provider/resource_keycloak_users_permissions.go

```diff
@@ -0,0 +1,272 @@
+				ForceNew: true,
+			},
+			"authorization_resource_server_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Resource server id representing the realm management client on which this permission is managed",
+			},
+			"view_scope_policy_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"manage_scope_policy_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"map_roles_scope_policy_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"manage_group_membership_scope_policy_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"impersonate_scope_policy_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"user_impersonated_scope_policy_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func setUsersScopePermissionPolicy(keycloakClient *keycloak.KeycloakClient, realmId string, scopeName string, policyId string) error {
+	usersPermissions, err := keycloakClient.GetUsersPermissions(realmId)
+	if err != nil {
+		return err
+	}
+
+	realmManagementClient, err := keycloakClient.GetOpenidClientByClientId(realmId, "realm-management")
+	if err != nil {
+		return err
+	}
+
+	permission, err := keycloakClient.GetOpenidClientAuthorizationPermission(realmId, realmManagementClient.Id, usersPermissions.ScopePermissions[scopeName].(string))
+	if err != nil {
+		return err
+	}
+
+	permission.Policies = []string{policyId}
+
+	return keycloakClient.UpdateOpenidClientAuthorizationPermission(permission)
+}
+
+func unsetUsersScopePermissionPolicy(keycloakClient *keycloak.KeycloakClient, realmId, scopeName string) error {
+	usersPermissions, err := keycloakClient.GetUsersPermissions(realmId)
+	if err != nil {
+		return err
+	}
+
+	realmManagementClient, err := keycloakClient.GetOpenidClientByClientId(realmId, "realm-management")
+	if err != nil {
+		return err
+	}
+
+	permission, err := keycloakClient.GetOpenidClientAuthorizationPermission(realmId, realmManagementClient.Id, usersPermissions.ScopePermissions[scopeName].(string))
+	if err != nil {
+		return err
+	}
+
+	permission.Policies = []string{}
+	err = keycloakClient.UpdateOpenidClientAuthorizationPermission(permission)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceKeycloakUsersPermissionsCreate(data *schema.ResourceData, meta interface{}) error {
+	return resourceKeycloakUsersPermissionsUpdate(data, meta)
+}
+
+func resourceKeycloakUsersPermissionsUpdate(data *schema.ResourceData, meta interface{}) error {
+	keycloakClient := meta.(*keycloak.KeycloakClient)
+
+	realmId := data.Get("realm_id").(string)
+
+	if data.Get("enabled").(bool) {
+		err := keycloakClient.EnableUsersPermissions(realmId)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := keycloakClient.DisableUsersPermissions(realmId)
+		if err != nil {
+			return err
+		}
+	}
+
+	viewScopePolicyId, ok := data.GetOkExists("view_scope_policy_id")
```

Comment:
Let's replace all of the instances of `data.GetOkExists` with `data.GetOk`, since `data.GetOkExists` is deprecated.